src/components/Auth/index.js

```diff
@@ -0,0 +1,98 @@
+const AuthSevice = require('./service');
+const UsersService = require('../User/service');
+const AuthValidation = require('./validation');
+const ValidationError = require('../../error/ValidationError');
+const AuthError = require('../../error/UnauthorizedError');
+const NotFoundError = require('../../error/NotFoundError');
+const ConflictError = require('../../error/ConflictError');
+
+async function signUp(req, res) {
+    const { error, value } = AuthValidation.signUp(req.body);
+
+    if (error) {
+        throw new ValidationError(error.details);
+    }
+    if (value.email !== value.email.toLowerCase()) throw new ValidationError('Email must be in lowercase');
+
+    const userExistens = await UsersService.findByEmail(value.email);
+    if (userExistens) throw new ConflictError('such user already exists');
+    const user = await UsersService.create(req.body);
+
+    return res.status(201).json({
+        data: user,
+    });
+}
+
+async function login(req, res) {
+    const { error, value } = AuthValidation.login(req.body);
+
+    if (value.email !== value.email.toLowerCase()) throw new ValidationError('Email must be in lowercase');
```

Comment:
strange validation, user can send email with upper case but we must save to lower case)